mmv1/third_party/validator/service_account.go

```diff
@@ -0,0 +1,146 @@
+// ----------------------------------------------------------------------------
+//
+//	This file is copied here by Magic Modules. The code for building up a
+//	storage bucket object is copied from the manually implemented
+//	provider file:
+//	third_party/terraform/resources/resource_storage_bucket.go
+//
+// ----------------------------------------------------------------------------
+package google
+
+import (
+	"fmt"
+	"reflect"
+)
+
+const ServiceAccountAssetType string = "iam.googleapis.com/ServiceAccount"
+
+func resourceConverterServiceAccount() ResourceConverter {
+	return ResourceConverter{
+		AssetType: ServiceAccountAssetType,
+		Convert:   GetServiceAccountCaiObject,
+	}
+}
+
+func GetServiceAccountCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	name, err := assetName(d, config, "//iam.googleapis.com/projects/{{project}}/serviceAccounts/{{unique_id}}")
+	if err != nil {
+		return []Asset{}, err
+	}
+	if obj, err := GetServiceAccountApiObject(d, config); err == nil {
+		return []Asset{{
+			Name: name,
+			Type: ServiceAccountAssetType,
+			Resource: &AssetResource{
+				Version:              "v1",
+				DiscoveryDocumentURI: "https://iam.googleapis.com/$discovery/rest",
+				DiscoveryName:        "ServiceAccount",
+				Data:                 obj,
+			},
+		}}, nil
+	} else {
+		return []Asset{}, err
+	}
+}
+
+func GetServiceAccountApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
+	obj := make(map[string]interface{})
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	descriptionProp, err := expandServiceAccountDescription(d.Get("description"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("description"); !isEmptyValue(reflect.ValueOf(descriptionProp)) && (ok || !reflect.DeepEqual(v, descriptionProp)) {
+		obj["description"] = descriptionProp
+	}
+
+	emailProp, err := expandServiceAccountDescription(d.Get("email"), d, config)
```

Comment:
Similar to the above comment, since email is an exported field, not a supported argument.

Reply:
Should the export to CAI contains:
1. Only attributes present in TF configuration file (.hcl file)
2. Or it can used also attributes present in the `tfplan` that have been generated server side but were not present as input during resource creation (but will present if this is an update)

Using 2. allows to reuse some policies defined in [config-validator](https://github.com/GoogleCloudPlatform/config-validator) which is using CAI export to evaluate GCP resources.
For example, [gcp_iam_restrict_service_account_creation_v1.yaml](https://github.com/GoogleCloudPlatform/policy-library/blob/main/policies/templates/gcp_iam_restrict_service_account_creation_v1.yaml) is using `email` attribute.

If we are following 1., then I will remove the code for `email`, `unique_id` and `name`. Let me know.

Reply:
The tfplan does not seem to be able to show value that is not in tf file. Do you have an example where tfplan converted from tf file show exported value?

Reply:
Yes attributes generated on server side are present in the TF Plan.

U can find it here (test case for existing service account):

https://github.com/GoogleCloudPlatform/magic-modules/blob/243c2d9b5e5c758d163356ca0d0eca9b5a825032/mmv1/third_party/validator/tests/data/example_service_account_update.tfplan.json#L53